### DISCUSSION TITLE
How can I render a surface texture inset from the edges of the window?

### DISCUSSION THREAD

**Discussion by tqwewe** [gfx-rs/wgpu#2533] [Q&A] [answered]
I'm looking to have bevy and tauri working together as an attempt at making a bevy editor, and I'd like a middle "viewport" to be rendered in bevy, while the surrounding area is html UI.

How can I render a surface texture (eg, bevy frames) not at (0, 0), but instead indended to make room for surounding UI?

![example](https://user-images.githubusercontent.com/16362377/157710447-46936641-d747-4405-8388-d14a3c394435.png)

*In the example, the red is the rendered frame, and the white is transparent and not rendered to.*

**Comment by cwfitzgerald** [comment, marked as answer]
You can set the viewport with https://docs.rs/wgpu/latest/wgpu/struct.RenderPass.html#method.set_viewport which will act like that part of the texture is what you're rendering to.

**Reply by tqwewe** [reply]
Thanks for the answer! I tried that but was getting an error saying the arguments I passed were incorrect. is there any examples on how to use it? I couldn't get the arguments correct.

**Reply by cwfitzgerald** [reply]
I don't have one off hand, but what parameters were you trying to use that didn't work?

**Reply by tqwewe** [reply]
I tried `.set_viewport(100.0, 100.0, 500.0, 500.0, 0.0, 10.0)`. I don't really understand what I'm doing with WGPU and kinda hacking it together, and wasn't sure what the min and max depth meant, so I just put random values of `0.0` and `10.0`.

**Reply by cwfitzgerald** [reply]
Ahhh the depth is almost certainly the problem, the maximum depth range is from 0.0 -> 1.0, so putting those values should work for you.
